src/imp/linux_raw/net/syscalls.rs

```diff
@@ -998,13 +1002,50 @@ pub(crate) mod sockopt {
         id: Timeout,
         timeout: Option<Duration>,
     ) -> io::Result<()> {
-        let timeout = match timeout {
+        let time = match timeout {
             Some(timeout) => {
                 if timeout == DURATION_ZERO {
                     return Err(io::Error::INVAL);
                 }
+                let mut timeout = __kernel_timespec {
+                    tv_sec: timeout.as_secs().try_into().unwrap_or(i64::MAX),
+                    tv_nsec: timeout.subsec_nanos() as _,
+                };
+                if timeout.tv_sec == 0 && timeout.tv_nsec == 0 {
+                    timeout.tv_nsec = 1;
+                }
```

Comment:
This is another nontrivial conversion logic that seems like it'd be best as a helper?

Reply:
Good idea; done..